ships/0031-shipwright-trigger.md

```diff
@@ -0,0 +1,343 @@
+
+The API also supports using selector labels to match Tekton Pipelines as well, giving a broader scope to select Pipelines to trigger Builds.
+
+To accomplish this task, Shipwright Trigger must watch over PipelineRun objects, and check if there are Builds in the inventory that should be activated. The business logic is recorded on the [PipelineRun Controller][triggerControllersPipelineRunController].
+
+## Trigger Actors
+
+This section explains what are the actions inside the Shipwright Trigger application, and how they interact with each other and with Kubernetes as well. Please consider the following image:
+
+![Shipwright Trigger Components](assets/0031-shipwright-trigger-components.drawio.png)
+
+In the next sections each actor is detailed.
+
+### Build Inventory
+
+The [inventory is the central component][triggerInventoryInventory] of Shipwright Trigger, it stores all the Build instances organized in a way that allows searching for types of triggers, depending on the Inventory client.
+
+For example, the WebHook Handler will always search for Builds based on the Git repository URL, the type of event (`Push` or `PullRequest`), and the branch names. In other hand, the other Controllers will query the inventory based on the `.objectRef` attribute instead.
+
+As you can see on the diagram above, almost all components are interacting with the Inventory using the specialized query methods `SearchForGit` and `SearchForObjectRef`.
+
+#### Shipwright Build Controller
+
+The Builds are added or removed from the Inventory through the [Build Controller][triggerControllersBuildController], responsible to reflect all Shipwright Build resources into the Inventory. On adding new entries, the Build is prepared for the subsequent queries.
+
+### Tekton Run Controller
+
+Watches for [Tekton Run instances][triggerControllersRunController] referencing Shipwright Builds, when a new instance is created it creates a new `BuildRun`. The controller also watches over the `BuildRun` instance, in order to reflect the status back to the Tekton Run parent.
+
+The Tekton Run instances are part of the Custom Task workflow, everytime Tekton finds a `TaskRef` resource outside of Tekton's scope, it creates a `Run` instance with the coordinates. In other words, to extend Tekton's functionality third party applications must watch and interact with those objects.
+
+### Tekton PipelineRun Controller
+
+The [controller for PipelineRun instances][triggerControllersPipelineRunController] is meant to react when a Pipeline reaches the desired status, so upon changes on the resource the controller checks on the inventory if there are triggers configured for the specific resource in question, in the desired status.
+
+Upon the creation of a `BuildRun` instance, the `PipelineRun` object is labeled for the controller to be able to avoid reprocessing.
+
+### WebHook Handler
```

Comment:
For the record, GitHub sends [Actions Workflow events](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#workflow_run) on the same WebHook channel. The design on this proposal is able to support such type of events as well.